planner/plansdk/plansdk.go

```diff
@@ -26,18 +26,17 @@ type PlanError struct {
 
 // Plan tells devbox how to start shells and build projects.
 type Plan struct {
-	ShellWelcomeMessage string `json:"shell_welcome_message,omitempty"`
-
 	NixOverlays []string `cur:"[...string]" json:"nix_overlays,omitempty"`
-
 	// DevPackages is the slice of Nix packages that devbox makes available in
 	// its development environment. They are also available in shell.
 	DevPackages []string `cue:"[...string]" json:"dev_packages"`
-
 	// RuntimePackages is the slice of Nix packages that devbox makes available in
 	// in both the development environment and the final container that runs the
 	// application.
 	RuntimePackages []string `cue:"[...string]" json:"runtime_packages"`
+	// packageExtensions is the slice of Nix packages that devbox wants to extend
+	// to include extra packages that needs global installation.
+	PackageExtensions []string `cue:"[...string]" json:"package_extensions,omitempty"`
```

Comment:
@mikeland86 for php, I wonder if we can use the same mechanism I did for nginx, in that we can create a folder `.devbox/gen/php/` and add it to PATH, symlink the content it to the nix store, the folder is not immutable, thus they can install packages there instead?

Reply:
I think you may be misunderstanding what these do in php. They are not installing packages, they are recompiling PHP with the appropriate extensions.  Some extensions are included by default but need to be "enabled". Other extensions are not included at all and require php to be built from source. 


There's no way to just add an extension into a folder. The actual source of PHP needs to change.

Reply:
We wouldn't need to recompile PHP if the store is mutable?